name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/GroupedAccountListView.java

```diff
@@ -0,0 +1,535 @@
+    {
+        setInput();
+
+        groupedAccounts.refresh();
+
+        if (groupedAccounts.getTree().getItemCount() > 0 && !items.isEmpty())
+            groupedAccounts.setSelection(new StructuredSelection(items.getFirst()));
+
+        expandNodes();
+    }
+
+    @Override
+    public void onModified(Object element, Object newValue, Object oldValue)
+    {
+        markDirty();
+    }
+
+    @Override
+    protected void addButtons(ToolBarManager toolBar)
+    {
+        addNewButton(toolBar);
+        addConfigButton(toolBar);
+    }
+
+    private void addNewButton(ToolBarManager toolBar)
+    {
+        toolBar.add(new DropDown(Messages.MenuCreateAndManageClientFilter, Images.PLUS, SWT.NONE,
+                        manager -> {
+                            manager.add(new LabelOnly(Messages.MenuCreateAndManageClientFilter));
+                            manager.add(new SimpleAction(Messages.LabelClientFilterNew, a -> {
+                                clientFilterMenu.createCustomFilter();
+                                groupedAccounts.refresh();
+                                // select the newly created account
+                                groupedAccounts.setSelection(new StructuredSelection(items.getFirst()));
+                            }));
+                            manager.add(new SimpleAction(Messages.LabelClientFilterManage, a -> {
+                                clientFilterMenu.editCustomFilter();
+                                groupedAccounts.refresh();
+                            }));
+                        }));
+    }
+
+    private void addConfigButton(final ToolBarManager toolBar)
+    {
+        toolBar.add(new DropDown(Messages.MenuShowHideColumns, Images.CONFIG, SWT.NONE,
+                        manager -> groupedAccountColumns.menuAboutToShow(manager)));
+    }
+
+    // //////////////////////////////////////////////////////////////
+    // top table: accounts
+    // //////////////////////////////////////////////////////////////
+
+    @Override
+    protected Control createBody(Composite parent)
+    {
+        Composite container = new Composite(parent, SWT.NONE);
+        TreeColumnLayout layout = new TreeColumnLayout();
+        container.setLayout(layout);
+
+        groupedAccounts = new TreeViewer(container, SWT.FULL_SELECTION);
+        final Tree tree = groupedAccounts.getTree();
+        tree.setHeaderVisible(true);
+        tree.setLinesVisible(true);
+
+        ColumnEditingSupport.prepare(groupedAccounts);
+        ColumnViewerToolTipSupport.enableFor(groupedAccounts, ToolTip.NO_RECREATE);
+        CopyPasteSupport.enableFor(groupedAccounts);
+
+        groupedAccountColumns = new ShowHideColumnHelper(GroupedAccountListView.class.getSimpleName() + "@top2", //$NON-NLS-1$
+                        getPreferenceStore(), groupedAccounts, layout);
+
+        Column column = new NameColumn("0", Messages.ClientEditorLabelClientMasterData, SWT.None, 100, getClient()); //$NON-NLS-1$
```

Comment:
Historically, the columns did not have a name. It was just the order as configured in the menu. That was a problem, because I could only add columns add the end. Therefore, some years back I added a name. To be backwards compatible, some columns still have the index as name. But now we can give proper names to the column.